java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1433,4 +1433,48 @@ public static ColumnVector timestampNanoSecondsFromBoxedLongs(Long... values) {
     return build(DType.TIMESTAMP_NANOSECONDS, values.length, (b) -> b.appendBoxed(values));
   }
 
+  /**
+   * Creates an empty column according to the data type.
+   *
+   * It will create all the nested columns by iterating all the children in the input
+   * type object 'colType'.
+   *
+   * The performance is not good, so use it carefully. We may want to move this implementation
+   * to the native once figuring out a way to pass the nested data type to the native.
```

Comment:
For handling nested types, wonder if we should just make this native.  As is this will create a lot of unnecessary Java objects, invoke a lot of degenerate copies, etc.

As one example of how to encode this for direct native creation, we can do a depth-first flattening of the data type into an array of integers that is encoded for each encountered type starting at the root type:
- LIST type emits the `DType` native ID for `LIST` followed by the recursive encoding of its child column type
- STRUCT emits the `DType` native ID for `STRUCT` followed by a child count and then the recursive encoding of each child column type in child order
- All other types emit the `DType` native ID for its type

We can then send this flattened integer array to native, and it can walk the type IDs to reconstruct the type hierarchy for the empty column.  I'm cool if we want to defer this to a followup issue.

Reply:
Agree.  Filed an issue for it. Here it is https://github.com/rapidsai/cudf/issues/8178